imap_processing/ena_maps/utils/spatial_utils.py

```diff
@@ -0,0 +1,270 @@
+    el_spacing: float = 0.5,
+) -> tuple[np.ndarray, np.ndarray, np.ndarray, np.ndarray]:
+    """
+    Build spatial bin boundaries for azimuth and elevation (input/output angles in deg).
+
+    Parameters
+    ----------
+    az_spacing : float, optional
+        The azimuth bin spacing in degrees (default is 0.5 degrees).
+    el_spacing : float, optional
+        The elevation bin spacing in degrees (default is 0.5 degrees).
+
+    Returns
+    -------
+    az_bin_edges : np.ndarray
+        Array of azimuth bin boundary values in degrees.
+    el_bin_edges : np.ndarray
+        Array of elevation bin boundary values in degrees.
+    az_bin_midpoints : np.ndarray
+        Array of azimuth bin midpoint values in degrees.
+    el_bin_midpoints : np.ndarray
+        Array of elevation bin midpoint values in degrees.
+    """
+    # Azimuth bins from 0 to 360 degrees.
+    az_bin_edges = np.arange(0, 360 + az_spacing, az_spacing)
+    az_bin_midpoints = az_bin_edges[:-1] + az_spacing / 2  # Midpoints between edges
+
+    # Elevation bins from -90 to 90 degrees.
+    el_bin_edges = np.arange(-90, 90 + el_spacing, el_spacing)
+    el_bin_midpoints = el_bin_edges[:-1] + el_spacing / 2  # Midpoints between edges
+
+    return az_bin_edges, el_bin_edges, az_bin_midpoints, el_bin_midpoints
+
+
+def build_solid_angle_map(
+    spacing: float, input_degrees: bool = True, output_degrees: bool = False
+) -> NDArray:
+    """
+    Build a solid angle map for a given spacing in degrees.
+
+    Parameters
+    ----------
+    spacing : float
+        The bin spacing in the specified units.
+    input_degrees : bool, optional
+        If True, the input spacing is in degrees
+        (default is True for radians).
+    output_degrees : bool, optional
+        If True, the output solid angle map is in square degrees
+        (default is False for steradians).
+
+    Returns
+    -------
+    solid_angle_grid : np.ndarray
+        The solid angle map grid in steradians (default) or square degrees.
+        First index is latitude/el, second index is longitude/az.
+    """
+    if input_degrees:
+        spacing = np.deg2rad(spacing)
+
+    if spacing <= 0:
+        raise ValueError("Spacing must be positive valued, non-zero.")
+
+    if not np.isclose((np.pi / spacing) % 1, 0):
+        raise ValueError("Spacing must divide evenly into pi radians.")
+
+    latitudes = np.arange(-np.pi / 2, np.pi / 2 + spacing, step=spacing)
+    sine_latitudes = np.sin(latitudes)
+    delta_sine_latitudes = np.diff(sine_latitudes)
+    solid_angle_by_latitude = np.abs(spacing * delta_sine_latitudes)
+
+    # Order ensures agreement with build_az_el_grid's order of tiling az/el grid.
+    solid_angle_grid = np.repeat(
+        solid_angle_by_latitude[np.newaxis, :], (2 * np.pi) / spacing, axis=0
+    )
+
+    if output_degrees:
+        solid_angle_grid *= (180 / np.pi) ** 2
+
+    return solid_angle_grid
+
+
+@typing.no_type_check
+def rewrap_even_spaced_el_az_grid(
```

Comment:
In your functions above this you have `az_el`. Are we able to keep the order consistent everywhere?
```suggestion
def rewrap_even_spaced_az_el_grid(
```

Reply:
Yes!